scripts/e2e-sudo

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+uid=`id -u`
+sudocmd="sudo"
+
+if [ $uid != 0 ]; then
```

Comment:
This bit here is duplicated in test-sudo.

It looks to me that you want to add a single sudo wrapper that doesn't have any knowledge about how tests are being run. Basically this check here and a simple `exec ${sudocmd} "$@"` below. Unless I'm missing something, the bit with `env -i` can go in e2e-test (this would take care of setting the environment correctly for those tests). If there's a reason this needs to be like this, it would be a nice gesture to future people looking at this code (including future you) to add a note here so that they don't have to figure it out by themselves.